### PR TITLE
fix backward compatibility of Regex

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Fix broken builds by fixing regex

### CODE REVIEW

File: src/inmanta/ast/constraint/expression.py [hunk 1]
```diff
@@ -331,7 +331,9 @@ class Regex(UnaryOperator):
 
     def __init__(self, op1: ExpressionStatement, op2: str):
         self.regex = re.compile(op2)
-        UnaryOperator.__init__(self, "regex", op1)
+        # This class used to inherit from BinaryOperator,
+        # We make this call to the shared super class to maintain backward compatibility
+        Operator.__init__(self, "regex", [op1, Literal(self.regex)])
```
Comment (sanderr) [inline src/inmanta/ast/constraint/expression.py:336]: Another option would be to revert the changes to `Regex` and make it a `BinaryOperator` again. If we interpret it more as the `matching` operator that makes some sense, and it would get rid of this unconventional structure.

**Review by sanderr — APPROVED** [review]
Perhaps we should also add a public interface to inspect the regex and its operand? `lsm` shouldn't access `Regex._arguments` (I can't believe I implemented that). We probably can't fix that now on the lsm side but we can at least add support already.

File: src/inmanta/ast/constraint/expression.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/ast/constraint/expression.py:336]: ok

**Review by sanderr — APPROVED** [review]
